src/Adapter/MSTest.CoreAdapter/Execution/TestClassInfo.cs

```diff
@@ -355,16 +365,16 @@ public void RunClassInitialize(TestContext testContext)
         [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Requirement is to handle all kinds of user exceptions and message appropriately.")]
         public string RunClassCleanup()
         {
-            if (this.ClassCleanupMethod is null && !this.BaseClassInitAndCleanupMethods.Any(p => p.Item2 != null))
+            if (this.ClassCleanupMethod is null && !this.BaseClassCleanupMethodsStack.Any())
```

Comment:
I can revert it back but it's basically an equivalent check because things don't get pushed into BaseClassCleanupMethodsStack unless Item2 in BaseClassInitAndCleanupMethods was not null.